pkg/openstack/instances_test.go

```diff
@@ -189,3 +190,56 @@ func TestSortNodeAddressesWithMultipleCIDRs(t *testing.T) {
 
 	executeSortNodeAddressesTest(t, addressSortOrder, want)
 }
+
+func Test_instanceIDFromProviderID(t *testing.T) {
+	type args struct {
+		providerID string
+	}
+	tests := []struct {
+		name           string
+		args           args
+		wantInstanceID string
+		wantRegion     string
+		wantErr        bool
+	}{
+		{
+			name: "it parses region & instanceID correctly from providerID",
+			args: args{
+				providerID: "openstack://us-east-1/testInstanceID",
+			},
+			wantInstanceID: "testInstanceID",
+			wantRegion:     "us-east-1",
+			wantErr:        false,
+		},
+		{
+			name: "it returns error in case of invalid providerID format",
+			args: args{
+				providerID: "openstack://us-east-1-testInstanceID",
+			},
+			wantInstanceID: "",
+			wantRegion:     "",
+			wantErr:        true,
+		},
+		{
+			name: "it appends openstack:// in case of missing protocal in providerID",
+			args: args{
+				providerID: "us-east-1/testInstanceID",
+			},
+			wantInstanceID: "testInstanceID",
+			wantRegion:     "us-east-1",
+			wantErr:        false,
+		},
+	}
```

Comment:
The most common case in OpenStack is like this: `openstack:///testInstanceID` Let's add it too.

Reply:
Acknowledged.

Reply:
I'd also test protocol injection in case the instance as an empty region. Can you please add a case like this?

```
providerID: "/testInstanceID",
```

Reply:
I have added the test-case

Reply:
I just checked, & I had named that test-case incorrectly. Here it is

Reply:
https://github.com/kubernetes/cloud-provider-openstack/blob/c4b34e1459587b652c305c6788dd1380e64c0dc2/pkg/openstack/instances_test.go#L215